REDCapUITweaker.php

```diff
@@ -7,24 +7,63 @@ class REDCapUITweaker extends \ExternalModules\AbstractExternalModule
 {
 	const SUBMIT_TYPES = [ 'record', 'continue', 'nextinstance',
 	                       'nextform', 'nextrecord', 'exitrecord' ];
+	const SUBMIT_DEFINE = 'record,nextinstance,nextform,continue';
 
 	private $customAlerts;
 
 
+
+
+
 	// Initialise module when enabled.
 	// Set default system settings and enable in all projects.
 
 	function redcap_module_system_enable()
 	{
-		$this->setSystemSetting( 'field-types-order', '1,2,7,5,6,11,3|4,8,10,9' );
-		$this->setSystemSetting( 'field-default-required', '1' );
-		$this->setSystemSetting( 'submit-option-tweak', '0' );
+		if ( $this->getSystemSetting( 'field-types-order' ) == '' &&
+		     $this->getSystemSetting( 'field-default-required' ) == '' &&
+		     $this->getSystemSetting( 'submit-option-tweak' ) == '' )
+		{
+			$this->setSystemSetting( 'field-types-order', '1,2,7,5,6,11,3|4,8,10,9' );
+			$this->setSystemSetting( 'field-default-required', '1' );
+			$this->setSystemSetting( 'submit-option-tweak', '0' );
+		}
+		elseif ( $this->getSystemSetting( 'submit-option-tweak' ) == '2' &&
+		         $this->getSystemSetting( 'submit-option-define' ) == '' )
+		{
+			$this->setSystemSetting( 'submit-option-define', self::SUBMIT_DEFINE );
+		}
 		$this->setSystemSetting( 'enabled', true );
 	}
 
 
 
 
+
+	// Remove the project setting for 'form submit options' if the system setting 'allow custom
+	// submit options' is not enabled.
+
+	function redcap_module_configuration_settings( $project_id, $settings )
+	{
+		if ( $project_id !== null && ! $this->getSystemSetting( 'submit-option-custom' ) )
```

Comment:
sorry - I may have got this backwards, but I wasn't expecting it to check ! $this - surely we DO want to replace setting if the submit-option-custom exists and is defined?

Reply:
The project setting _submit-option_ will be displayed as defined in the config.json file by default.
If the system setting _submit-option-custom_ is *not* enabled, then the _submit-option_ project setting should be hidden.
This is the intended behaviour.